setup.py

```diff
@@ -1,34 +1,55 @@
-from setuptools import setup
+from setuptools import setup, find_packages
+from pip.req import parse_requirements
+from pip.download import PipSession
 
+from os import path
 
+
+# Lists of requirements and dependency links which are needed during runtime, testing and setup
+install_requires = []
+tests_require = []
+dependency_links = []
+
+# Inject requirements from requirements.txt into setup.py
+requirements_file = parse_requirements(path.join('requirements', 'requirements.txt'), session=PipSession())
+for req in requirements_file:
+    install_requires.append(str(req.req))
+    if req.link:
+        dependency_links.append(str(req.link))
+
+# Inject test requirements from requirements_test.txt into setup.py
+requirements_test_file = parse_requirements(path.join('requirements', 'requirements_test.txt'), session=PipSession())
+for req in requirements_test_file:
+    tests_require.append(str(req.req))
+    if req.link:
+        dependency_links.append(str(req.link))
+
```

Comment:
I'm not fan of this method. I'd really rather have a setup.py that doesn't depend on external files.